clippy_lints/src/operators/numeric_arithmetic.rs

```diff
@@ -15,8 +15,10 @@ pub struct Context {
     const_span: Option<Span>,
 }
 impl Context {
-    fn skip_expr(&mut self, e: &hir::Expr<'_>) -> bool {
-        self.expr_id.is_some() || self.const_span.map_or(false, |span| span.contains(e.span))
+    fn skip_expr<'tcx>(&mut self, cx: &LateContext<'tcx>, e: &'tcx hir::Expr<'_>) -> bool {
+        self.expr_id.is_some()
+            || self.const_span.map_or(false, |span| span.contains(e.span))
+            || is_from_proc_macro(cx, e)
```

Comment:
Does this fixes affect `float_arithmetic`? If so, the tests for `float_arithmetic` and changelog entry should be updateded.

Reply:
Modified to only check `integer_arithmetic`